commit 0858608b52a30495aa9718af4dece524dd567716
Author: Felix Stupp <felix.stupp@banananet.work>
Date:   Tue Jun 3 07:48:44 2025 +0000

    WIP radv prefix timeout

diff --git a/src/libsystemd-network/sd-radv.c b/src/libsystemd-network/sd-radv.c
index f241929ad5..c9fb3f842d 100644
--- a/src/libsystemd-network/sd-radv.c
+++ b/src/libsystemd-network/sd-radv.c
@@ -548,6 +548,47 @@ void sd_radv_remove_prefix(
         if (!ra || !prefix)
                 return;
 
+        // need current flags for prefix for potential fade out
+        bool opt_found = false;
+        sd_ndisc_option *opt;
+        SET_FOREACH(opt, ra->options) {
+                if (opt->type != SD_NDISC_OPTION_PREFIX_INFORMATION)
+                        continue;
+
+                if (!in6_addr_prefix_intersect(&opt->prefix.address, opt->prefix.prefixlen, prefix, prefixlen))
+                        continue; /* no intersection */
+
+                if (opt->prefix.prefixlen != prefixlen)
+                        continue; /* not the same prefix */
+
+                opt_found = true;
+                break;
+        }
+
+        // only trigger fade out when prefix information was found
+        if (opt_found) {
+                uint64_t current_time = now(CLOCK_REALTIME);
+                // TODO convert to being an option
+                uint64_t two_hours_usec = 2 * USEC_PER_HOUR;
+                // TODO decide to what limits error checking in this void function is feasable
+                // see https://github.com/systemd/systemd/pull/34244#issuecomment-2933872831
+                ndisc_option_set_prefix(
+                                &ra->options,
+                                opt->prefix.flags,
+                                prefixlen,
+                                prefix,
+                                // valid lifetime
+                                two_hours_usec,
+                                // preferred lifetime
+                                0,
+                                // valid until
+                                current_time + two_hours_usec,
+                                // preferred until
+                                current_time);
+                return;
+        }
+
+        // if no prefix found, trigger removal anyway, just in case, preserving previous behavior
         ndisc_option_remove(ra->options,
                             &(sd_ndisc_option) {
                                     .type = SD_NDISC_OPTION_PREFIX_INFORMATION,
